### ISSUE TITLE
Break Homebrew teleport package into client and server

### ISSUE THREAD

**Issue by sover02** [gravitational/teleport#3002]
Small feature request, related to: https://github.com/gravitational/teleport/issues/514

On MacOS, `brew install teleport` is fantastic.  We can use it to keep our users up to date with the latest `tsh` client. However, from a security standpoint it's hard for us to justify having every user install the `teleport` server binary (and `tctl`) and can lead to some interesting conversations if scanners pick up the software on all client machines, even if the daemon is not running.

I know users can download and unpack the tarball manually, but I don't think it is reasonable for us to ask our large user base to keep track of and manually update a piece of software every quarter – we want to keep this package-managed.

TL;DR: Can we break the homebrew packages into something like `teleport-server` (with `teleport` and `tctl`) and `teleport-client` (`tsh`)?

Thanks

**Comment by klizhentas** [comment]
makes sense to me, let me schedule it for 4.2

**Comment by webvictim** [comment]
I think the Brew package is actually maintained externally - we didn't write it, someone else did. I'd probably suggest raising that issue against the maintainer in the brew repo.

**Comment by webvictim** [comment]
https://github.com/Homebrew/homebrew-core/pull/44377 looks to be the PR that merged Teleport 4.0.9 into brew.

I think essentially for brew to look at doing this, we (Gravitational) would need to start building and hosting separate binary artefacts for `['tctl', 'teleport']` and `['tsh']`. That's probably a bigger decision we'd need to look at internally.

**Comment by benarent** [comment]
I would +1 this and do like the intent. When I first started it wasn't 100% clear that you user `tsh` and not teleport.  ( as per our user manual ) https://gravitational.com/teleport/docs/user-manual/  

It would be great if we had quick links for each OSS, for how to quickly setup the tsh-client for teams who have Teleport setup in their Org.  

Per 

> This would increase the number of downloadable packages on the website from 11 to 24 and basically make it impossible to find the download you're looking for. 

With other requirements, we'll need to address soon anyway. Maybe we could start with just breaking up our Brew package for Mac users first.

**Comment by sover02** [comment]
Perhaps a small link at the bottom of the download page:

>[Looking for client-only packages?](#)

Definitely an expert on UX, but even just a deep-link that administrators can send their users to would be valuable. If we want to convince several hundred engineers in our org that this is a viable replacement to the SSH they are familiar with, we really need to be able to reduce friction anywhere possible. 

Being able to send our users to a deep-link with just the binaries they need (and the auto-updates to go with) would go a long way to making the transition as seamless as possible.

**Comment by webvictim** [comment]
Depends on #3040

**Comment by bcg62** [comment]
Generally speaking `brew install` only covers applications that homebrew itself complies and hosts. They do not allow pulling pre-complied binary artifacts. In order to do this `brew cask install` is used for 3rd party, self hosted pkg/dmg packages

**Comment by zmb3** [comment]
As mentioned, the homebrew package is built by the community, so this  request should be submitted upstream.
